sqlglot/parser.py

```diff
@@ -1917,6 +1920,28 @@ def _parse_convert(self):
             to = None
         return self.expression(exp.Cast, this=this, to=to)
 
+    def _parse_substring(self):
+        this = self._parse_term()
+
+        # Postgres supports the form: substring(string [from int] [for int])
+        # https://www.postgresql.org/docs/9.1/functions-string.html @ Table 9-6
+        matching_delimiter = {
+            TokenType.FROM: TokenType.FOR,
+            TokenType.COMMA: TokenType.COMMA,
+        }
+
+        start = None
+        length = None
+
+        if self._match_set((TokenType.FROM, TokenType.COMMA)):
+            delim = self._prev
+            start = self._parse_term()
+
+            if self._match(matching_delimiter[delim.token_type]):
+                length = self._parse_term()
+
+        return self.expression(exp.Substring, this=this, start=start, length=length)
```

Comment:
maybe if parse substring fails you can retreat?

Reply:
Ah, I see. I'm not sure if that's simple to do with the current implementation, because once we match against the function's parser we lose the option of executing the else branch.

Reply:
that’s ok